File: javascript/easy/removeLinkedListElements.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} val
 * @return {ListNode}
 */
var removeElements = function(head, val) {
    while(head != null && head.val == val) {
        head = head.next;
    }
    if(head == null) return null;
    
    let current = head, runner = head.next;
    while(runner != null) {
        if(runner.val === val) {
            current.next = runner.next;
        } else {
            current = current.next;
        }
        runner = runner.next;
    }
    
    return head;
};

File: ruby/easy/remove_linked_list_elements.rb
# Definition for singly-linked list.
# class ListNode
#     attr_accessor :val, :next
#     def initialize(val)
#         @val = val
#         @next = nil
#     end
# end

# @param {ListNode} head
# @param {Integer} val
# @return {ListNode}
def remove_elements(head, val)
    while head && head.val == val
        head = head.next
    end
    current = head
    while(current && current.next)
        if(current.next.val == val)
            current.next = current.next.next
        else 
            current = current.next
        end
    end
    return head
end